specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2021-04-01-preview/cosmos-db.json

```diff
@@ -9139,7 +9153,69 @@
       }
     },
     "ServiceResource": {
-      "description": "Service cluster response resource.",
+      "description": "Data transfer response resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique resource identifier of the ARM resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The service name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of Azure resource."
+        },
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "The properties of a Data Transfer service.",
+          "$ref": "#/definitions/ServiceProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "DataTransferServiceResource": {
+      "description": "Data transfer response resource.",
+      "type": "object",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The unique resource identifier of the ARM resource."
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The service name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The type of Azure resource."
+        },
+        "properties": {
+          "type": "object",
+          "x-ms-client-flatten": true,
+          "description": "The properties of a Data Transfer service.",
+          "$ref": "#/definitions/ServiceProperties"
+        }
+      },
+      "required": [
+        "properties"
+      ],
+      "x-ms-azure-resource": true
+    },
+    "SqlDedicatedGatewayServiceResource": {
```

Comment:
Instead of creating two different resource (SqlDedicatedGatewayServiceResource and DataTransferServiceResource)
we should create two different Properties( SqlDedicatedGatewayServiceResource and DataTransferServiceResourceProperties) and use discriminator in ServiceResource.properties.properties object